src/pages/Insights/InsightViewPage.js

```diff
@@ -29,6 +30,14 @@ const InsightViewPage = ({
   isDesktop,
   isLoggedIn
 }) => {
+  const [bannerRef, setBannerRef] = useState()
+
+  const measuredRef = useCallback(node => {
```

Comment:
`useRef` instead of `useCallback`

Reply:
useRef not suitable, because it mutate ref, but I need to update data after getting ref. [Documentation](https://reactjs.org/docs/hooks-reference.html#useref)

Reply:
And what's the problem with using the reference? `checkVisibility` still we able to get all necessary data for calculations

Reply:
because when checkVisibility is called - ref is empty. I tried useRef and after change to useCallback, because was problem with it

Reply:
Don't know what was wrong for you, but it's working fine for me
![image](https://user-images.githubusercontent.com/25135650/56656379-0b2ee780-669e-11e9-8e75-5d63b19cb3d3.png)

Reply:
it work because you didn't delete timer with `50` maybe

Reply:
I'll try

Reply:
Oh, so you wanted for it to appear from the beginning. It's okay if it will show up after the scroll

Reply:
what about mobile? I don't use scroll event here

Reply:
The problem is in the component structure. `BannerMobile` is somehow aware of the logic which is applied to the `BannerDesktop`. That's an incorrect usage of the composition.
`AnonBannerSticky` should render `BannerMobile` if it's a mobile device and `AnonBannerStickyDesktop` if it's a desktop. `AnonBannerStickyDesktop`would be a component, which contains necessary logic for the conditional rendering of the `BannerMobile` desktop.

Reply:
hm.. you mean that on mobile I should show sticky banner always when insight opened? Without checking position of static banner and it's OK if they will be both on screen?

Reply:
@Partysun Didn't we decided to show banner by default after insight was opened on a mobile device?